### ISSUE TITLE
Cosmos provider: Input string was not in a correct format when data saved.

### ISSUE THREAD

**Issue by adamlepkowski** [aspnet/EntityFrameworkCore#16153]
When Entity Framework Core is used to operate with Cosmos DB provider to create a new entity exception is thrown (Input string was not in a correct format.), and the entity is stored inside a Cosmos DB.

```
Exception message:  Input string was not in a correct format.
Stack trace:
System.AggregateException: One or more errors occurred. (Exception has been thrown by the target of an invocation.) ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.FormatException: Input string was not in a correct format.
   at System.Number.ParseDouble(ReadOnlySpan`1 value, NumberStyles options, NumberFormatInfo numfmt)
   at Microsoft.Azure.Cosmos.CosmosResponseMessageHeaders.set__requestCharge(String value)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.RuntimePropertyInfo.SetValue(Object obj, Object value, BindingFlags invokeAttr, Binder binder, Object[] index, CultureInfo culture)
   at System.Reflection.RuntimePropertyInfo.SetValue(Object obj, Object value, Object[] index)
   at Microsoft.Azure.Cosmos.CosmosMessageHeadersInternal.Add(String headerName, String value)
   at Microsoft.Azure.Cosmos.Extensions.ToCosmosResponseMessage(DocumentServiceResponse response, CosmosRequestMessage requestMessage)
   at Microsoft.Azure.Cosmos.Handlers.TransportHandler.SendAsync(CosmosRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Azure.Cosmos.Handlers.AbstractRetryHandler.ExecuteHttpRequestAsync(Func`1 callbackMethod, Func`3 callShouldRetry, Func`3 callShouldRetryException, CancellationToken cancellationToken)
   at Microsoft.Azure.Cosmos.Handlers.AbstractRetryHandler.SendAsync(CosmosRequestMessage request, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at Microsoft.Azure.Cosmos.ExecUtils.<>c__DisplayClass3_0`1.<ProcessResourceOperationAsync>b__0(Task`1 task)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.EntityFrameworkCore.Cosmos.Storage.Internal.CosmosClientWrapper.CreateItemOnceAsync(DbContext _, ValueTuple`2 parameters, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Cosmos.Storage.Internal.CosmosClientWrapper.CreateItemOnce(DbContext context, ValueTuple`2 parameters)
   at Microsoft.EntityFrameworkCore.Storage.ExecutionStrategy.ExecuteImplementation[TState,TResult](Func`3 operation, Func`3 verifySucceeded, TState state)
   at Microsoft.EntityFrameworkCore.Storage.ExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.Cosmos.Storage.Internal.CosmosClientWrapper.CreateItem(String containerId, JToken document)
   at Microsoft.EntityFrameworkCore.Cosmos.Storage.Internal.CosmosDatabaseWrapper.Save(IUpdateEntry entry)
   at Microsoft.EntityFrameworkCore.Cosmos.Storage.Internal.CosmosDatabaseWrapper.SaveChanges(IList`1 entries)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(IList`1 entriesToSave)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at ConsoleApp2.Program.AddObject() in C:\Users\alepkowski\source\repos\ConsoleApp2\ConsoleApp2\Program.cs:line 61
   at ConsoleApp2.Program.Main(String[] args) in C:\Users\alepkowski\source\repos\ConsoleApp2\ConsoleApp2\Program.cs:line 40
```

### Steps to reproduce
Copy the below code to a console application project and execute it.

```c#
public class Consortium
    {
        public Guid ConsortiumId { get; set; }
        public string Name { get; set; }
    }

    public class ExpanseContext : DbContext
    {
        public DbSet<Consortium> Consortia { get; set; }
        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            optionsBuilder.UseCosmos(
                "url",
                "super_secret_key",
                "db_name");

        }
    }

    class Program
    {
        static void Main(string[] args)
        {
            CreateDB();

            AddObject();
        }

        private static void CreateDB()
        {
            using (var context = new ExpanseContext())
            {
                context.Database.EnsureCreated();
            }
        }

        private static void AddObject()
        {
            var consortium = new Consortium
            {
                ConsortiumId = Guid.NewGuid(),
                Name = "Test"
            };
            using (var context = new ExpanseContext())
            {
                context.Consortia.Add(consortium);
                context.SaveChanges();
            }
        }
    }
```

### Further technical details
EF Core version: 3.0.0-preview5.19227.1 and 3.0.0-preview6.19304.10
Database Provider: Microsoft.EntityFrameworkCore.Cosmos
IDE: Visual Studio 2017 15.9.2

**Comment by AndriySvyryd** [comment]
Fixed in c2a39d700963f71c0035fa297aae6ec430578951

**Comment by AndriySvyryd** [comment]
@adamlepkowski This should work in preview8 or nightly builds. Let us know if it still fails for you.
